src/Interfaces/AbstractZonotope.jl

```diff
@@ -525,3 +525,52 @@ function constraints_list(Z::AbstractZonotope{N}; check_full_rank::Bool=true
     end
     return constraints
 end
+
+"""
+    rectify(Z::AbstractZonotope{N}) where {N}
+
+Concrete rectification of a zonotopic set.
+
+### Input
+
+- `Z` -- zonotope
+
+### Output
+
+The zonotope obtained by rectifying `Z`.
+
+### Algorithm
+
+This function implements [Theroem 3.1, 1].
+
+[1] *Singh, G., Gehr, T., Mirman, M., Püschel, M., & Vechev, M. (2018). Fast
+and effective robustness certification. In Advances in Neural Information
+Processing Systems (pp. 10802-10813).*
+"""
+function rectify!(Z::AbstractZonotope{N}) where {N}
+    c = Z.center
+    G = Z.generators
+    n, m = size(G)
+    H = overapproximate(Z, Hyperrectangle)
+    Gnew = zeros(N, n, n)
+
+    @inbounds for i in 1:n
+        lx, ux = H.center[i] - H.radius[i], H.center[i] + H.radius[i]
+        if !_leq(lx, zero(N))
+            nothing
+        elseif _leq(ux, zero(N)) || isapproxzero(lx)
+            c[i] = zero(N)
+            G[i, :] = zeros(N, m)
+        else
+            λ = ux / (ux - lx)
+            μ = - λ * lx / 2
+            c[i] = c[i] * λ + μ
+            G[i, :] = G[i, :] .* λ
+            Gnew[i, i] = μ
+        end
+    end
+    Z = Zonotope(c, hcat(G, Gnew))
+    return remove_zero_generators(Z)
```

Comment:
We had some discussion about this in the past and I think we concluded that we do not remove zero generators (a user can do that manually).

In the other case, it would be more efficient to create the `Zonotope` only once.
```suggestion
    Z = Zonotope(c, remove_zero_columns(hcat(G, Gnew)))
    return Z
```

Reply:
I agree with the proposed change. In fact i would only check  `Gnew`, if this is possible

```julia
hcat(G, remove_zero_columns(Gnew))
```

Reply:
In fact it would be a better algorithm to just store the indices `i` where  `Gnew` is different from zero.